File: apps/spa-3ncount3r/src/view-models/encounter-creature.view-model.ts
import { CreatureViewModel } from "./creature.view-model";

export class EncounterCreatureViewModel extends CreatureViewModel {
    id: string = '';
    hitpointMax: number = 0;
    currentHitpoints: number = 0;
    initiative: number = 0;
    isPlayerCharacter: boolean = false;
    isActive: boolean = true;
}

File: backends/DDD.orch3strator/ViewModels/DnD5e/EncounterCreatureViewModel.cs
using System.Text.Json.Serialization;

namespace DDD.orch3strator.ViewModels.DnD5e
{
  public class EncounterCreatureViewModel : CreatureViewModel
  {
    [JsonPropertyName("id")]
    public string Id { get; set; }

    [JsonPropertyName("hitpointMax")]
    public int HitpointMax { get; set; }

    [JsonPropertyName("currentHitpoints")]
    public int CurrentHitpoints { get; set; }

    [JsonPropertyName("temporaryHitpoints")]
    public int TemporaryHitpoints { get; set; }

    [JsonPropertyName("initiative")]
    public int Initiative { get; set; }

    [JsonPropertyName("isPlayerCharacter")]
    public bool IsPlayerCharacter { get; set; } = false;

    [JsonPropertyName("isActive")]
    public bool IsActive { get; set; } = true;
  }
}
